bin/improver-percentile

```diff
@@ -107,6 +107,8 @@ def main():
         fast_percentile_method = True
         if np.ma.is_masked(cube.data):
             fast_percentile_method = False
+        elif np.ma.isMaskedArray(cube.data):
```

Comment:
This could do with a bit of explanation. I found some explanation in https://github.com/scipy/scipy/pull/3998#issuecomment-55651288, which says that `is_masked` will return False, if the array is a masked array but none of the points within the array are masked.

Reply:
Added in 6011555